### PR TITLE
chore(ci): tfv head and tfv base depends on tpg-test

### PR DESCRIPTION
**Pull request by iyabchen**
- tfv-head and tfv-base depends on tpg-test
- move all tfv related steps lower, and renumber the step 

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/gcb-generate-diffs.yml [hunk 1]
```diff
@@ -157,7 +194,7 @@ steps:
         - COMMIT_SHA=$COMMIT_SHA
         - PR_NUMBER=$_PR_NUMBER
       args:
-          - $_PR_NUMBER
+          - $_PR_NUMBER
```
Comment (melinath) [inline .ci/gcb-generate-diffs.yml:197]:
```suggestion
          - $_PR_NUMBER
```

File: .ci/gcb-generate-diffs.yml [hunk 2]
```diff
@@ -110,41 +110,78 @@ steps:
...
+      args:
           - 'ga'
           - $_PR_NUMBER
+          - $COMMIT_SHA
+          - $BUILD_ID
+          - $PROJECT_ID
+          - GoogleCloudPlatform/magic-modules
+          - "16"  # Build step
+
+    - name: 'gcr.io/graphite-docker-images/gcb-terraform-vcr-tester'
```
Comment (melinath) [inline .ci/gcb-generate-diffs.yml:156]: This step needs to happen after `diff`. Error:

```
invalid build: invalid .steps field: build step https://github.com/GoogleCloudPlatform/magic-modules/pull/17 - "gcb-tpg-vcr-test" depends on "diff", which has not been defined
```

https://github.com/GoogleCloudPlatform/magic-modules/pull/7517/checks?check_run_id=12205682779

**Review by melinath — CHANGES_REQUESTED** [review]

File: .ci/gcb-generate-diffs.yml (hunk 2 above)
Comment (iyabchen) [reply .ci/gcb-generate-diffs.yml]: Done

**Review by melinath — CHANGES_REQUESTED** [review]
After seeing how this will actually impact the time for making the diff available to users - I think maybe we should hold off on implementing the reordering change, at least for now. We could instead just mark the steps as allowed failures so that they don't halt the build. Note: at this point, it looks likely that https://github.com/GoogleCloudPlatform/magic-modules/pull/7446 will land soon, possibly today.

**Comment by iyabchen** [comment]
Dropping this, and created #7526 for allow failure

**Pull request closed without merging**